pkg/tracer/tracer.go

```diff
@@ -715,39 +722,57 @@ func (m *FlowFetcher) ReadRingBuf() (ringbuf.Record, error) {
 // LookupAndDeleteMap reads all the entries from the eBPF map and removes them from it.
 // TODO: detect whether BatchLookupAndDelete is supported (Kernel>=5.6) and use it selectively
 // Supported Lookup/Delete operations by kernel: https://github.com/iovisor/bcc/blob/master/docs/kernel-versions.md
-func (m *FlowFetcher) LookupAndDeleteMap(met *metrics.Metrics) map[ebpf.BpfFlowId][]ebpf.BpfFlowMetrics {
+func (m *FlowFetcher) LookupAndDeleteMap(met *metrics.Metrics) map[ebpf.BpfFlowId]model.BpfFlowContent {
 	if !m.lookupAndDeleteSupported {
 		return m.legacyLookupAndDeleteMap(met)
 	}
 
 	flowMap := m.objects.AggregatedFlows
 
 	iterator := flowMap.Iterate()
-	var flows = make(map[ebpf.BpfFlowId][]ebpf.BpfFlowMetrics, m.cacheMaxSize)
+	var flows = make(map[ebpf.BpfFlowId]model.BpfFlowContent, m.cacheMaxSize)
 	var ids []ebpf.BpfFlowId
 	var id ebpf.BpfFlowId
-	var metrics []ebpf.BpfFlowMetrics
+	var baseMetrics ebpf.BpfFlowMetrics
+	var additionalMetrics []ebpf.BpfAdditionalMetrics
 
 	// First, get all ids and don't care about metrics (we need lookup+delete to be atomic)
-	for iterator.Next(&id, &metrics) {
+	for iterator.Next(&id, &baseMetrics) {
 		ids = append(ids, id)
 	}
 
 	count := 0
 	// Run the atomic Lookup+Delete; if new ids have been inserted in the meantime, they'll be fetched next time
 	for i, id := range ids {
 		count++
-		if err := flowMap.LookupAndDelete(&id, &metrics); err != nil {
+		if err := flowMap.LookupAndDelete(&id, &baseMetrics); err != nil {
 			if i == 0 && errors.Is(err, cilium.ErrNotSupported) {
 				log.WithError(err).Warnf("switching to legacy mode")
 				m.lookupAndDeleteSupported = false
 				return m.legacyLookupAndDeleteMap(met)
 			}
-			log.WithError(err).WithField("flowId", id).Warnf("couldn't delete flow entry")
+			log.WithError(err).WithField("flowId", id).Warnf("couldn't lookup/delete flow entry")
 			met.Errors.WithErrorName("flow-fetcher", "CannotDeleteFlows").Inc()
 			continue
 		}
-		flows[id] = metrics
+		flowPayload := model.BpfFlowContent{}
+		flowPayload.AccumulateBase(&baseMetrics)
+
+		// Fetch additional metrics; ids are without direction and interface
+		shorterID := id
+		shorterID.Direction = 0
+		shorterID.IfIndex = 0
```

Comment:
is that related to hack in the bpf code where they aren't set ?

Reply:
correct